src/osd/PG.cc

```diff
@@ -2864,13 +2866,19 @@ epoch_t PG::peek_map_epoch(ObjectStore *store,
     values.clear();
     keys.insert(ek);
     store->omap_get_values(META_COLL, legacy_infos_oid, keys, &values);
-    assert(values.size() == 1);
+    if (values.size() < 1) {
```

Comment:
> Do we need to ensure this is not an accident data file loss?

I'm not sure how to tell.

But, let's say the infos object *is* missing but shouldn't be.. in that 
case, all PGs will be ignored and the osd will come up (vs previously we 
would crash).  There is a corner case where this is bad: the osd in 
question had the only copy of the latest write, peering probed it, and got 
a does not exist response and silently reverts to an older copy.  
Something similar happens from the pg import/export tests currently (see 
#12687).  I don't think there is a way to tell, though...